src/info/infoBackup.c

```diff
@@ -459,6 +468,52 @@ infoBackupDataAdd(const InfoBackup *this, const Manifest *manifest)
     FUNCTION_LOG_RETURN_VOID();
 }
 
+/**********************************************************************************************************************************/
+void
+infoBackupDataAnnotationSet(const InfoBackup *this, const String *const backupLabel, const KeyValue *annotationKv)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INFO_BACKUP, this);
+        FUNCTION_TEST_PARAM(STRING, backupLabel);
+        FUNCTION_TEST_PARAM(KEY_VALUE, annotationKv);
+    FUNCTION_TEST_END();
+
+    ASSERT(this != NULL);
+    ASSERT(infoBackupLabelExists(this, backupLabel));
+
+    MEM_CONTEXT_BEGIN(lstMemContext(this->pub.backup))
+    {
+        InfoBackupData *infoBackupData = lstFind(this->pub.backup, &backupLabel);
+
+        KeyValue *tmpAnnotationKv = kvNew();
+
+        if (infoBackupData->backupAnnotation != NULL)
+            tmpAnnotationKv = varKv(infoBackupData->backupAnnotation);
+
+        const VariantList *const annotationKeyList = kvKeyList(annotationKv);
+
+        for (unsigned int keyIdx = 0; keyIdx < varLstSize(annotationKeyList); keyIdx++)
```

Comment:
I'm not sure why we need to create a copy here? Why not just modify the existing KV? (Yes, you need to remove the const in InfoBackupData.)

Reply:
Hm, everything is stored as Variant so I needed to transform into KV to modify it. Perhaps there's indeed an easier way to achieve it, but I didn't found it yet ^^'

Reply:
Worked in this in f88abc26925ad670. I misread this a bit -- there was not a copy being made (though there needed to be earlier) but it was leaking a KV. Made general improvements to make the function clearer.